backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EventRepository.java

```diff
@@ -125,7 +127,19 @@ private static String addHqlConditions(String hql, Set<UUID> bundleIds, Set<UUID
         if (checkEndpointType || checkInvocationResult) {
             List<String> subQueryConditions = new ArrayList<>();
             if (checkEndpointType) {
-                subQueryConditions.add("nh.endpointType IN (:endpointTypes)");
+                List<String> subQueryEndpointTypes = new ArrayList<>();
+                Set<EndpointType> basicTypes = endpointTypes.stream().filter(c -> c.getSubType() == null).map(CompositeEndpointType::getType).collect(Collectors.toSet());
+                Set<CompositeEndpointType> compositeTypes = endpointTypes.stream().filter(c -> c.getSubType() != null).collect(Collectors.toSet());
```

Comment:
With the current code, this is executed 4 times when `EventResource#getEvents` is called. Could you move it to `EventResource` and pass the basic and composite types as method params to `EventRepository`?

Reply:
My [other comment](https://github.com/RedHatInsights/notifications-backend/pull/1183#discussion_r855374697) may change things a bit here.

Reply:
Sure

Reply:
done